### PR TITLE
Rest of Classes Page

### PR DESCRIPTION
**Pull request by Kathirilango**

### CODE REVIEW

**Comment by Kathirilango** [comment]
Screen Shot 2020-11-24 at 12 26 21 AM

File: cs130_front/src/components/Inviter/Inviter.js [hunk 1]
```diff
@@ -10,10 +10,15 @@ export default function Inviter(props) {
   function sendSelection(group){
     setGroup(group);
   }
-
+  
   return(
+    props.invitedUser == "" ?
```
Comment (shirlyfang) [inline cs130_front/src/components/Inviter/Inviter.js:15]: props.user == null ?

File: cs130_front/src/components/Inviter/Inviter.js [hunk 2]
```diff
@@ -10,10 +10,15 @@ export default function Inviter(props) {
...
+    <div className="small-rounded-box">
+      <Text size="24px" weight="800">Select a user to invite to a group</Text>
+    </div>
+    :
     <div className="small-rounded-box">
-      <Text size="24px" weight="800">Invite to group</Text>
+      <Text size="24px" weight="800">Invite {props.invitedUser.split(' ')[0]} to group</Text>
```
Comment (shirlyfang) [inline cs130_front/src/components/Inviter/Inviter.js:21]:
```
<Text size="24px" weight="800">Invite {props.user.name.split(' ')[0]} to group</Text>
```

File: cs130_front/src/components/UserList/UserItem.js [hunk 3]
```diff
@@ -20,7 +20,7 @@ const userItem = (props) => (
             }
         </div>
         <div className="col-centered">
-          {props.optionalElement ? <Button onClick={ () => props.optionalClick(props.user)}> Invite to group </Button> : null}
+          {props.optionalElement ? <Button onClick={ () => props.optionalClick(props.user.name)}> Invite to group </Button> : null}
```
Comment (shirlyfang) [inline cs130_front/src/components/UserList/UserItem.js:23]: go back to original

File: cs130_front/src/components/views/ClassesPage/ClassesPage.js [hunk 4]
```diff
@@ -7,21 +7,28 @@ import ListingList from '../../Listings/ListingList';
...
+  ]
+  const currentGroup = { id: 123, name: "DM Squad", courseName: "Calculus", members: members, day: "friday", time: "4:30pm" }
+  
   const [mainTitleState, setMainTitle] = useState(classes[0].name);
   const [mainListingsDefault, setMainListingsDefault] = useState(classes[0].listings);
   const [mainListings, setMainListings] = useState(classes[0].listings);
-  const [currentTab, setCurrentTab] = useState("Study Buddies")
+  const [currentTab, setCurrentTab] = useState("Study Buddies");
   const [input, setInput] = useState('');
+  const [invitedUser, setInvitedUser] = useState('');
```
Comment (shirlyfang) [inline cs130_front/src/components/views/ClassesPage/ClassesPage.js:31]: const [invitedUser, setInvitedUser] = useState(null);

File: cs130_front/src/components/views/ClassesPage/ClassesPage.js [hunk 5]
```diff
@@ -50,11 +57,18 @@ function ClassesPage(props) {
     if (name == "Listings"){
       setMainListings(mainListingsDefault);
     }
+    if (name == "Study Buddies"){
+      setInvitedUser("");
```
Comment (shirlyfang) [inline cs130_front/src/components/views/ClassesPage/ClassesPage.js:61]: setInvitedUser(null);

File: cs130_front/src/components/views/ClassesPage/ClassesPage.js [hunk 6]
```diff
@@ -66,12 +80,13 @@ function ClassesPage(props) {
             
               <ListingCreator user={user} course_name={mainTitleState} items={items}></ListingCreator>
               :
-              <Inviter user={user} items={items} handleGroupInvite={handleGroupInvite}/> 
+              <Inviter user={user} items={items} handleGroupInvite={handleGroupInvite} invitedUser={invitedUser}/>
```
Comment (shirlyfang) [inline cs130_front/src/components/views/ClassesPage/ClassesPage.js:83]:
```
<Inviter user={invitedUser} items={items} handleGroupInvite={handleGroupInvite}/> 
```

**Review by shirlyfang — CHANGES_REQUESTED** [review]
Most of my review deals with passing the actual user object around to the inviter instead of just the name. because of this we only need the user prop for inviter and we set it to listen to the invitedUser state var when a click on "invite to group happens". otherwise, lgtm.

**Review by shirlyfang — APPROVED** [review]

**Pull request merged**